src/dotnet/Chat.Contracts/ChatsBackendExt.cs

```diff
@@ -140,15 +140,49 @@ public static class ChatsBackendExt
 
     public static async IAsyncEnumerable<ApiArray<Chat>> BatchChanged(
         this IChatsBackend chatsBackend,
-        bool includePeerChats,
         long minVersion,
         long maxVersion,
         ChatId lastId,
         int batchSize,
         [EnumeratorCancellation] CancellationToken cancellationToken)
     {
         while (!cancellationToken.IsCancellationRequested) {
-            var chats = await chatsBackend.ListChanged(includePeerChats, minVersion, maxVersion, lastId, batchSize, cancellationToken)
+            var chats = await chatsBackend.ListChanged(new ChangedChatsQuery {
+                    MinVersion = minVersion,
+                    MaxVersion = maxVersion,
+                    LastId = lastId,
+                    Limit = batchSize,
+                }, cancellationToken)
+                .ConfigureAwait(false);
+            if (chats.Count == 0)
+                yield break;
+
+            yield return chats;
+
+            var last = chats[^1];
+            lastId = last.Id;
+            minVersion = last.Version;
+        }
+    }
+
+    public static async IAsyncEnumerable<ApiArray<Chat>> BatchChangedGroups(
```

Comment:
What do you need both BatchChanged and BatchChangedGroups methods? They looks identical.

Reply:
Removed the first one since it's not even used